### ISSUE TITLE
Lineage Tag preventing model validation

### ISSUE THREAD

**Issue by samfeinb** [microsoft/Analysis-Services#111]
I followed several other now closed issues regarding the lineage messing up version control.  Basically, when i am running validation step on pbix and workspace data sets, the warnings/updates dialog runs, then a popup explains An Object with lineage tax ................ a;ready exists in the collection and ALM goes blank.    I used Tabular editor script provided by otykier in issue https://github.com/otykier/TabularEditor/issues/819 to null out linneage tags in both the source and target , saved and ran with every option i can think of in the options dialog (especially newly implemented "Consider lineage tag when comparing" .  Script below.  
foreach(var m in Model.AllMeasures) m.LineageTag = null;
foreach(var l in Model.AllLevels) l.LineageTag = null;
foreach(var h in Model.AllHierarchies) h.LineageTag = null;
foreach(var c in Model.AllColumns) c.LineageTag = null;
foreach(var t in Model.Tables) t.LineageTag = null;

When i manually looked for lineage tags in every object i can think of i still can't find the offender.  I'd appreciate some thoughts.  There are major changes in the source like pulling from blob rather than sql along with many new changes.  I did notice that gen2 is not supported in ALM yet so I'm using the blob connector.

Thank you for all the great efforts and tools.  they are 1000% essential... though the prevalence of these tools compared to other enterprise level bi tools is a bit daunting.  It seems that power bi is mostly thought of as an office type tool rather than a full on enterprise BI service.  Any other thoughts on this are welcome.

**Comment by christianwade** [comment]
Hi, if you are using the script to null out the lineage tag, then please leave "consider lineage tag when comparing" = false. This will allow them to show up in the definitions and hopefully get blanked out. 

For the error that says "an object with lineage tag ..... already exists", it might be worth scripting out the whole model in SSMS and doing a search on the whole metadata file for that lineage tag or part of it.

Let me know if you solve it. That's all I can think of right now ...

**Comment by samfeinb** [comment]
thank you for reply!  I used consider lineage tag when comparing false....but no luck.  I also pulled the bim from the source and target into visual studio and notepad++.  The lineage tag guids that its finding are not in either!  My thought is that it seems possible that there is a "secret" lineage tag under the covers somewhere.  I have no idea where or how to get to it, but Daniel's script above must be missing it somehow.

**Comment by christianwade** [comment]
Would I be able to reproduce the issue in my local environment if you sent me file (like BIM/PBIX/T files)?

**Comment by samfeinb** [comment]
I have resolved.  It seems the error was caused by me reusing queries. I updated all the names of the power queries to include actions taken (Connect, clean, transform, merge, proxy) instead of my previous number scheme.  I also removed the lineage tags via script and had to reprocess the model for that to take effect.  I then used ALM and that removed the previous errors. 

I guess the matching is done on query names and tries to take into account lineage tags. Perhaps a post some time about the roles they play in the version control process might be helpful to me and others.

Thank you for responses!  I feel a special sense of connection after seeing so much of your content and being able to speak psuedo directly to you.  Thank you!
